diff --git a/ReactCommon/React-Fabric.podspec b/ReactCommon/React-Fabric.podspec
index 0644c97ba22b5c74bf14e8e5efa0fc81227f78ae..bda664b68ea47e4b366c9e08932d8d89a5d4a431 100644
--- a/ReactCommon/React-Fabric.podspec
+++ b/ReactCommon/React-Fabric.podspec
@@ -33,7 +33,9 @@ Pod::Spec.new do |s|
   s.source                 = source
   s.source_files           = "dummyFile.cpp"
   s.pod_target_xcconfig = { "USE_HEADERMAP" => "YES",
-                            "CLANG_CXX_LANGUAGE_STANDARD" => "c++17" }
+                            "CLANG_CXX_LANGUAGE_STANDARD" => "c++17",
+                            "DEFINES_MODULE" => "YES"
+                            }
 
   if ENV['USE_FRAMEWORKS']
     s.header_mappings_dir     = './'
